File: javascript/pangram/pangram.js
// test suit is looking to import
// import { isPangram } from './pangram';
//  which looks like
// isPangram is the function and './pangram' is the file

export const isPangram = (sentence) => {
  if (sentence.length == 0) {return false;}

  const ALPHABET = 'abcdefghijklmnopqrstuvwxyz';

  // make an object with keys of all the letters
  const alphaObject = {}
  ALPHABET.split('').forEach(letter => {
    alphaObject[letter] = 0
  });

  // add 1 to the value of each letter present in sentence
  const cleanedSentence = sentence.toLowerCase().replace(/\W/g,'')
  cleanedSentence.split('').forEach(letter => {
    alphaObject[letter] ++
  });

  // are there any letters which have a value of 0?
  // return the inverse.
  return !Object.values(alphaObject).includes(0);
}

File: ruby/pangram/pangram.rb
class Pangram

  def self.pangram? phrase
    phrase = phrase.upcase.chars
    alphabet = ("A".."Z").to_a
    if (phrase & alphabet).length == alphabet.length
      true
    else
      false
    end

  end
end

module BookKeeping
  VERSION = 6 # Where the version number matches the one in the test.
end


# class Pangram
#   class << self
#
#     def pangram? phrase
#       phrase = phrase.upcase.split ""
#       alphabet = ("A".."Z").to_a
#
#       if (phrase & alphabet).length == 26
#         return true
#       end
#       false
#
#     end
#   end
# end































# class Pangram
#
#   # initialize phrase
#   # end
#
#
#   def self.pangram? string
#
#     if string == ''
#       return false
#     end
#
#     alpha_array = ("a".."z").to_a
#     string_array = string.downcase.split('')
#
#     # alpha_array.each do |letter|
#     #   if (string_array.index(letter) == nil )
#     #     return false
#     #   end
#     # end
#
#     alpha_array.all? { |letter| string_array.include?(letter) }
#
#
#   end
#
# end
